lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -1938,17 +1947,30 @@ public function NewObjectExpression()
     }
 
     /**
-     * NewObjectArg ::= ScalarExpression | "(" Subselect ")"
+     * NewObjectArg ::= ScalarExpression | NamedScalarExpression | "(" Subselect ")"
      *
      * @return mixed
      */
-    public function NewObjectArg()
+    public function NewObjectArg(/* bool $namedArgAlreadyParsed = false */)
     {
+        $namedArgAlreadyParsed = func_get_args()[0] ?? false;
```

Comment:
Do we actually need this parameter? Couldn't `NewObjectExpression()` just check if the kind of argument it receives is supported at its position?

Reply:
I think this should work (but idk how to feel abt it): when parsing a `NewObjectExpression`, we parse the current one, consume the tokens, then parse the next one (if there is any) without consuming it, and throw if it's not named while the current one is. It would work without the need of the parameter, but i feel like it's kinda doing too much ? also we're parsing twice all the arguments while it's not really needed (idk if it's really impactant in terms of perf). wdyt ?

Reply:
Hello @derrabus, I didn't want to ping earlier since it was holiday season; what do you think about the approach describe above ? did you have something else in mind ?